nifi/nifi-nar-bundles/nifi-flume-bundle/nifi-flume-processors/src/test/java/org/apache/nifi/processors/flume/FlumeSinkProcessorTest.java

```diff
@@ -0,0 +1,165 @@
+            logger.error(vr.toString());
+            Assert.assertTrue(vr.toString().contains("is invalid because Sink Type is required"));
+        }
+
+        // non-existent class
+        results = new HashSet<>();
+        runner.setProperty(FlumeSinkProcessor.SINK_TYPE, "invalid.class.name");
+        runner.enqueue(new byte[0]);
+        pc = runner.getProcessContext();
+        if (pc instanceof MockProcessContext) {
+            results = ((MockProcessContext) pc).validate();
+        }
+        Assert.assertEquals(1, results.size());
+        for (ValidationResult vr : results) {
+            logger.error(vr.toString());
+            Assert.assertTrue(vr.toString().contains("is invalid because unable to load sink"));
+        }
+
+        // class doesn't implement Sink
+        results = new HashSet<>();
+        runner.setProperty(FlumeSinkProcessor.SINK_TYPE, AvroSource.class.getName());
+        runner.enqueue(new byte[0]);
+        pc = runner.getProcessContext();
+        if (pc instanceof MockProcessContext) {
+            results = ((MockProcessContext) pc).validate();
+        }
+        Assert.assertEquals(1, results.size());
+        for (ValidationResult vr : results) {
+            logger.error(vr.toString());
+            Assert.assertTrue(vr.toString().contains("is invalid because unable to create sink"));
+        }
+
+        results = new HashSet<>();
+        runner.setProperty(FlumeSinkProcessor.SINK_TYPE, NullSink.class.getName());
+        runner.enqueue(new byte[0]);
+        pc = runner.getProcessContext();
+        if (pc instanceof MockProcessContext) {
+            results = ((MockProcessContext) pc).validate();
+        }
+        Assert.assertEquals(0, results.size());
+    }
+
+
+    @Test
+    public void testNullSink() throws IOException {
+        TestRunner runner = TestRunners.newTestRunner(FlumeSinkProcessor.class);
+        runner.setProperty(FlumeSinkProcessor.SINK_TYPE, NullSink.class.getName());
+        FileInputStream fis = new FileInputStream("src/test/resources/testdata/records.txt");
+        Map<String, String> attributes = new HashMap<>();
+        attributes.put(CoreAttributes.FILENAME.key(), "records.txt");
+        runner.enqueue(fis, attributes);
+        runner.run();
+        fis.close();
+    }
+
+    @Test
+    public void testBatchSize() throws IOException {
+        TestRunner runner = TestRunners.newTestRunner(FlumeSinkProcessor.class);
+        runner.setProperty(FlumeSinkProcessor.SINK_TYPE, NullSink.class.getName());
+        runner.setProperty(FlumeSinkProcessor.BATCH_SIZE, "1000");
+        runner.setProperty(FlumeSinkProcessor.FLUME_CONFIG,
+            "tier1.sinks.sink-1.batchSize = 1000\n");
+        for (int i = 0; i < 100000; i++) {
+          runner.enqueue(String.valueOf(i).getBytes());
+        }
+        runner.run();
+    }
+    
+    @Test
+    public void testHdfsSink() throws IOException {
+        File destDir = new File("target/hdfs");
```

Comment:
That's not how I read the javadoc:
> The TemporaryFolder Rule allows creation of files and folders that should be deleted when the test method finishes (whether it passes or fails). Whether the deletion is successful or not is not checked by this rule. No exception will be thrown in case the deletion fails.

In particular the phrase "Whether the deletion is successful or not is not checked by this rule." worries me. If you're confident this works as expected I'm ok with it. These tests don't write a lot of data to the filesystem but I worry about cases where a test crashes mid run and could potentially leave a large amount of temporary data on the user's system.